tests/components/bring/test_todo.py

```diff
@@ -0,0 +1,394 @@
+                "spec": "",
+                "uuid": "b5d0790b-5f32-4d5c-91da-e29066f167de",
+            },
+        ],
+        BringItemOperation.REMOVE,
+    )
+
+
+async def test_delete_items_exception(
+    hass: HomeAssistant,
+    bring_config_entry: MockConfigEntry,
+    mock_bring_client: AsyncMock,
+) -> None:
+    """Test delete item."""
+
+    bring_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(bring_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert bring_config_entry.state is ConfigEntryState.LOADED
+    mock_bring_client.batch_update_list.side_effect = BringRequestException
+    with pytest.raises(
+        HomeAssistantError,
+        match=re.escape("Failed to delete 1 item(s) from Bring! list"),
+    ):
+        await hass.services.async_call(
+            TODO_DOMAIN,
+            TodoServices.REMOVE_ITEM,
+            service_data={ATTR_ITEM: "b5d0790b-5f32-4d5c-91da-e29066f167de"},
+            target={ATTR_ENTITY_ID: "todo.einkauf"},
+            blocking=True,
+        )
+
+
+async def test_send_notification(
+    hass: HomeAssistant,
+    bring_config_entry: MockConfigEntry,
+    mock_bring_client: AsyncMock,
+) -> None:
+    """Test send bring push notification."""
+
+    bring_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(bring_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert bring_config_entry.state is ConfigEntryState.LOADED
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_PUSH_NOTIFICATION,
+        service_data={
+            ATTR_NOTIFICATION_TYPE: "GOING_SHOPPING",
+        },
+        target={ATTR_ENTITY_ID: "todo.einkauf"},
+        blocking=True,
+    )
+
+    mock_bring_client.notify.assert_called_once_with(
+        "e542eef6-dba7-4c31-a52c-29e6ab9d83a5",
+        BringNotificationType.GOING_SHOPPING,
+        None,
+    )
+
+
+async def test_send_notification_exception(
+    hass: HomeAssistant,
+    bring_config_entry: MockConfigEntry,
+    mock_bring_client: AsyncMock,
+) -> None:
+    """Test send bring push notification with exception."""
+
+    bring_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(bring_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert bring_config_entry.state is ConfigEntryState.LOADED
+    mock_bring_client.notify.side_effect = BringRequestException
+    with pytest.raises(
+        HomeAssistantError,
+        match="Failed to send push notification for bring due to a connection error, try again later",
+    ):
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_PUSH_NOTIFICATION,
+            service_data={
+                ATTR_NOTIFICATION_TYPE: "GOING_SHOPPING",
+            },
+            target={ATTR_ENTITY_ID: "todo.einkauf"},
+            blocking=True,
+        )
+
+
+async def test_send_notification_service_validation_error(
```

Comment:
This doesn't seem to be a todo thing

Reply:
It is an entity ~~service~~ action that is registered for the todo platform, so I assumed this to be the right place but i can move it somewhere else